commands/intakecommands.py

```diff
@@ -0,0 +1,89 @@
+from commands2 import ParallelCommandGroup
+from commands.elevatorsetting import (
+    ElevatorBottomPosition,
+    ElevatorAmpPosition,
+    ElevatorTopPosition,
+)
+from commands.intakesetting import (
+    FloorIntake,
+    StageIntake,
+    FeedIntakeToShooter,
+    EjectInAmp,
+    HoldIntakeAtHandoff,
+    EjectInTrap,
+)
+from subsystems.elevatorsubsystem import ElevatorSubsystem
+from subsystems.intakesubsystem import IntakeSubsystem
+from subsystems.shootersubsystem import ShooterSubsystem
+
+
+class GroundIntake(ParallelCommandGroup):
+    def __init__(self, elevator: ElevatorSubsystem, intake: IntakeSubsystem):
+        ParallelCommandGroup.__init__(
+            ElevatorBottomPosition(elevator), FloorIntake(intake)
+        )
+        self.setName(__class__.__name__)
+
+
+class DefaultIntake(ParallelCommandGroup):
+    def __init__(self, elevator: ElevatorSubsystem, intake: IntakeSubsystem):
+        ParallelCommandGroup.__init__(
+            ElevatorBottomPosition(elevator), HoldIntakeAtHandoff(intake)
+        )
+        self.setName(__class__.__name__)
+
+
+class PrepareAmp(ParallelCommandGroup):
+    def __init__(self, elevator: ElevatorSubsystem, intake: IntakeSubsystem):
+        ParallelCommandGroup.__init__(
+            ElevatorAmpPosition(elevator), StageIntake(intake)
+        )
+        self.setName(__class__.__name__)
+
+
+class PrepareTrap(ParallelCommandGroup):
+    def __init__(self, elevator: ElevatorSubsystem, intake: IntakeSubsystem):
+        ParallelCommandGroup.__init__(
+            ElevatorTopPosition(elevator), StageIntake(intake)
+        )
+        self.setName(__class__.__name__)
+
+
+class ScoreTrap(ParallelCommandGroup):
+    def __init__(self, elevator: ElevatorSubsystem, intake: IntakeSubsystem):
+        commands = []
+
+        if (
+            intake.state == intake.IntakeState.Staging or intake.IntakeState.Trap
+        ) and elevator.state == elevator.ElevatorState.TopPosition:
+            commands = [EjectInTrap(intake)]
+
+        ParallelCommandGroup.__init__(*commands)
+        self.setName(__class__.__name__)
+
+
+# scores amp or feeds to shooter depending on state
+class DynamicScore(ParallelCommandGroup):
+    def __init__(
```

Comment:
since this is on init, it only runs once
be aware that this likely isn't what you want, instead being in execute where it calls these subcommands and determines next steps from bot action

Reply:
the init is for adding the commands that you want to run, the parallel command group runs the execute and the other stuff from the original commands

Reply:
if so, consider that the init function is only called at the first time the command is instance, as in right when bot code inits, so the stateful nature you are trying to do isn't going to be called since it happens after bot init